### ISSUE TITLE
how can I avoid the overfitting problem during training fastText model?

### ISSUE THREAD

**Issue by xiaoxiaofengzi** [facebookresearch/fastText#506]
The correct rate of the training set is 90%，but the correct rate of the test set is 60%, how can I avoid the overfitting problem? For example, which params can I change?
`fastText.train_supervised('data.txt',lr=0.05,epoch=40,wordNgrams=3,vervose=2,minCount=1,label='__label__')`
Thank you very much!

**Comment by FrostbiteNg** [comment]
if you reduce the "epoch" value, the effect is equivalent to earlystop. I think this helps your problem.

**Comment by EdouardGrave** [comment]
Hi @xiaoxiaofengzi,

As @FrostbiteNg said, one way to control overfitting is to reduce the number of epochs (parameter `epoch`). You can also try to use a smaller learning rate (parameter `lr`), a smaller dimension (parameter `dim`) or a smaller length of word ngrams (parameter `wordNgrams`).

Best,
Edouard.
